client/src/redux/reducers/sourcesReducer.js

```diff
@@ -28,20 +29,45 @@ const initialState = {
   }
 };
 
+const selectedIndex = function(state, source) {
+  return _.findIndex(state.persist.selectedSources, nextSelected => {
+    return nextSelected.id === _.get(source, 'id');
+  });
+};
+
 const sourcesReducer = function(state = initialState, action) {
   switch (action.type) {
     // Persist
-    case sourcesTypes.SOURCES_SELECTED:
-      return helpers.setStateProp(
-        'persist',
-        {
-          selectedSources: action.selectedSources
-        },
-        {
-          state,
-          reset: false
-        }
-      );
+    case sourcesTypes.SELECT_SOURCE:
+      // Do nothing if it is already selected
+      if (selectedIndex(state, action.source) !== -1) {
+        return state;
+      }
+
+      action.source.selected = true;
+
+      const addedSelections = Object.assign({}, state.persist, {
+        selectedSources: [...state.persist.selectedSources, action.source]
+      });
+      return Object.assign({}, state, { persist: addedSelections });
```

Comment:
Note: This is incredibly similar to the removed code with an addition, just long form

Reply:
You are correct sir. I will adjust to use the setStateProp